keel-core/src/main/kotlin/com/netflix/spinnaker/keel/veto/unhappy/UnhappyVeto.kt

```diff
@@ -49,7 +49,8 @@ class UnhappyVeto(
     val resourceId = resource.id
     val application = resource.application
 
-    if (diffFingerprintRepository.diffCount(resourceId) <= maxDiffCount(resource)) {
+    // maxDiff count represents the number of times we're allowed to see a diff and take action to try and fix it.
+    if (diffFingerprintRepository.actionTakenCount(resourceId) <= maxDiffCount(resource)) {
```

Comment:
using action taken count instead of diff count here to prepare for future changes

Reply:
Can you say more about what those future changes will be?

Reply:
https://github.com/spinnaker/keel/pull/1563/files#diff-43eecffdd336c15071ef33731c885473R168-R169